app/src/main/java/pm/gnosis/heimdall/ui/transactions/review/ReviewTransactionViewModel.kt

```diff
@@ -0,0 +1,187 @@
+package pm.gnosis.heimdall.ui.transactions.review
+
+import io.reactivex.Observable
+import io.reactivex.Single
+import io.reactivex.android.schedulers.AndroidSchedulers
+import pm.gnosis.heimdall.data.repositories.SignaturePushRepository
+import pm.gnosis.heimdall.data.repositories.TokenRepository
+import pm.gnosis.heimdall.data.repositories.TransactionData
+import pm.gnosis.heimdall.data.repositories.TransactionExecutionRepository
+import pm.gnosis.heimdall.data.repositories.models.SafeTransaction
+import pm.gnosis.heimdall.helpers.AddressHelper
+import pm.gnosis.heimdall.helpers.SignatureStore
+import pm.gnosis.heimdall.ui.transactions.review.viewholders.AssetTransferViewHolder
+import pm.gnosis.heimdall.ui.transactions.review.viewholders.GenericTransactionViewHolder
+import pm.gnosis.heimdall.utils.emitAndNext
+import pm.gnosis.model.Solidity
+import pm.gnosis.models.Wei
+import pm.gnosis.svalinn.accounts.base.models.Signature
+import pm.gnosis.svalinn.common.utils.DataResult
+import pm.gnosis.svalinn.common.utils.ErrorResult
+import pm.gnosis.svalinn.common.utils.Result
+import pm.gnosis.svalinn.common.utils.mapToResult
+import pm.gnosis.utils.asEthereumAddressString
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class ReviewTransactionViewModel @Inject constructor(
+    private val addressHelper: AddressHelper,
+    private val executionRepository: TransactionExecutionRepository,
+    private val signaturePushRepository: SignaturePushRepository,
+    private val signatureStore: SignatureStore,
+    private val tokenRepository: TokenRepository
+) : ReviewTransactionContract() {
+
+    private var safe: Solidity.Address? = null
```

Comment:
I'd declare this as a `lateinit` var. The exception thrown by not initializing the field would be caught up in an `Observable` and would remove a lot of `!!`

Reply:
will change with the next PR